internal/builtinfunctions/functions_test.go

```diff
@@ -819,3 +822,194 @@ func Test_floatToFormattedString_success(t *testing.T) {
 		}
 	}
 }
+
+func expectedOutputBindConstants(repInp any, items []any) []any {
+	return []any{
+		map[string]any{builtinfunctions.CombinedObjPropertyItemName: items[0], builtinfunctions.CombinedObjPropertyConstantName: repInp},
+		map[string]any{builtinfunctions.CombinedObjPropertyItemName: items[1], builtinfunctions.CombinedObjPropertyConstantName: repInp},
+		map[string]any{builtinfunctions.CombinedObjPropertyItemName: items[2], builtinfunctions.CombinedObjPropertyConstantName: repInp},
+	}
+}
+
+func Test_bindConstants(t *testing.T) {
+	items := []any{
+		map[string]any{"loop_id": 1},
+		map[string]any{"loop_id": 2},
+		map[string]any{"loop_id": 3},
+	}
+	repeatedInputsMap := map[string]any{
+		"a": "A", "b": "B",
+	}
+	repeatedInputsInt := 2
+	repeatedInputsRegexPattern := regexp.MustCompile("p([a-z]+)ch")
+	testItems := map[string]any{
+		"combine with map":   repeatedInputsMap,
+		"combine with list":  items,
+		"combine with int":   repeatedInputsInt,
+		"combine with regex": repeatedInputsRegexPattern,
+	}
+	functionToTest := builtinfunctions.GetFunctions()["bindConstants"]
+
+	for testName, testValue := range testItems {
+		tvLocal := testValue
+		t.Run(testName, func(t *testing.T) {
+			output, err := functionToTest.Call([]any{items, tvLocal})
+			assert.NoError(t, err)
+			assert.Equals(t, output.([]any), expectedOutputBindConstants(tvLocal, items))
+		})
+	}
+
+	t.Run("no items in input list", func(t *testing.T) {
+		output, err := functionToTest.Call([]any{[]any{}, repeatedInputsMap})
+		assert.NoError(t, err)
+		assert.Equals(t, output.([]any), []any{})
+	})
+}
+
+func defaultPropertySchema(t schema.Type) *schema.PropertySchema {
+	return schema.NewPropertySchema(t, nil, false, nil, nil, nil, nil, nil)
+}
+
+func joinStrs(s1, s2 string) string {
+	return strings.Join([]string{s1, s2}, builtinfunctions.CombinedObjIDDelimiter)
+}
+
+// TestHandleTypeSchemaCombine tests that the error cases for invalid
+// and valid input cases create the expected behavior in the returned
+// error and returned type and type name, respectively.
```

Comment:
The new wording is better. It seems like a run-on sentence. The second part, starting with "in the returned", can likely be its own sentence.

Reply:
Rerun your parser, @jaredoconnell:  there's one verb in the main clause (`tests`) and there's one verb in the subordinate clause (`create`); `in the returned` modifies the object (`behavior`) of the verb in the subordinate clause -- that phrase cannot be the start of a separate sentence, since there is no subject or verb associated with it.

That said, if you cannot easily make sense of the words we should probably rewrite it, but it's not incorrect grammar.  🙂

Reply:
It is correct grammar, yes. But there's so much going on that it should likely be re-worded.

Reply:
i just removed that second part